packages/account-v2/src/modules/FinancialAssessment/FinancialAssessmentForm.tsx

```diff
@@ -0,0 +1,228 @@
+            cfdTradingExperience,
+            cfdTradingFrequency,
+            forexTradingExperience,
+            forexTradingFrequency,
+            otherInstrumentsTradingExperience,
+            otherInstrumentsTradingFrequency,
+        }),
+    };
+
+    type TFinancialAssessmentFormValues = typeof initialValues;
+
+    const onSubmit = (handleSubmit: (e?: React.FormEvent<HTMLFormElement> | undefined) => void) => () => {
+        if (hasTradingExperience && isTradingExperienceNotComplete) {
+            setIsConfirmModalOpen(true);
+        } else {
+            handleSubmit();
+        }
+    };
+
+    const handleFormSubmit = async (values: TFinancialAssessmentFormValues) => {
+        updateFinancialAssessment({
+            account_turnover: values.accountTurnover,
+            education_level: values.educationLevel,
+            employment_industry: values.employmentIndustry,
+            employment_status: values.employmentStatus,
+            estimated_worth: values.estimatedWorth,
+            income_source: values.incomeSource,
+            net_income: values.netIncome,
+            source_of_wealth: values.sourceOfWealth,
+            ...(shouldHideOccupation(values.employmentStatus) ? {} : { occupation: values.occupation }),
+            ...(hasTradingExperience && {
+                binary_options_trading_experience: values.binaryOptionsTradingExperience,
+                binary_options_trading_frequency: values.binaryOptionsTradingFrequency,
+                cfd_trading_experience: values.cfdTradingExperience,
+                cfd_trading_frequency: values.cfdTradingFrequency,
+                forex_trading_experience: values.forexTradingExperience,
+                forex_trading_frequency: values.forexTradingFrequency,
+                other_instruments_trading_experience: values.otherInstrumentsTradingExperience,
+                other_instruments_trading_frequency: values.otherInstrumentsTradingFrequency,
+            }),
+        });
+    };
+
+    const onSuccessActionButton = () => {
+        history.push(redirect ? redirectUrl : ACCOUNT_V2_ROUTES.ProofOfAddress);
+        if (redirect) {
+            window.sessionStorage.removeItem('config.platform');
+        }
+    };
+
+    if (activeAccount?.is_virtual) return <DemoMessage />;
+
+    if (isFinancialAssessmentLoading) return <Loader />;
+
+    if (fetchError)
+        return <ActionScreen icon={<DerivLightIcPoaLockIcon width={128} />} title={fetchError.error.message} />;
+
+    if (isMobile && isAuthenticated && !isEUCountry && isFinancialAssessmentUpdateSuccess)
+        return (
+            <ActionScreen
+                actionButtons={
+                    <Button onClick={onSuccessActionButton}>
+                        {redirect ? `Back to ${redirectPlatformName}` : 'Continue'}
+                    </Button>
+                }
+                description={redirect ? '' : 'Let’s continue with providing proofs of address and identity.'}
```

Comment:
why are we showing "Let’s continue with providing proofs of address and identity." when isAuthenticated is true?
Client is already authenticated here. isnt?

Reply:
My mistake is was supposed to be !authenticated